packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2074,10 +2078,12 @@ function commitMutationEffectsOnFiber(
         // TODO: This is a temporary solution that allowed us to transition away
         // from React Flare on www.
         if (flags & Ref) {
-          if (current !== null) {
+          if (!offscreenSubtreeWasHidden && current !== null) {
             safelyDetachRef(finishedWork, finishedWork.return);
           }
-          safelyAttachRef(finishedWork, finishedWork.return);
+          if (!offscreenSubtreeIsHidden) {
+            safelyAttachRef(finishedWork, finishedWork.return);
+          }
```

Comment:
@kassens, I think this is the correct behavior.

You were right that `!offscreenSubtreeWasHidden` should only be scoped to `safelyDetachRef`, and we were missing a `!offscreenSubtreeIsHidden` check around `safelyAttachRef`.

Reply:
Are you sure about this behavior? We don't check for Offscreen anywhere else that we attach.

Reply:
> I'm not sure why `ScopeComponent` requires `safelyAttachRef` to be here, [...]

I traced the origin of that comment (which was harder than I expected due to the reconciler forking and unforking that happened) and found this: https://github.com/facebook/react/pull/19264

> To unblock internal progression on replacing React Flare, let's instead move resolution of Scope API refs to the mutation phase, so we can attach event listeners to scopes before the layout phase begins.

I don't fully understand the context, but I figured that it would be helpful to know its origin.

Reply:
After writing unit tests for this bug, I've confirmed that these changes are actually *not* needed to prevent `<Scope ref={X}>` from invoking `X` in `<Activity mode="hidden">`.

I've dropped these changes for now.

Reply:
I confirmed that `Scope` was previously completely ignoring any parent `Activity` that declared the subtree as hidden.